model/transformers/dffml_model_transformers/qa/qa_model.py

```diff
@@ -0,0 +1,1006 @@
+            optimizer.load_state_dict(
+                torch.load(
+                    os.path.join(
+                        self.parent.config.model_name_or_path, "optimizer.pt"
+                    )
+                )
+            )
+            scheduler.load_state_dict(
+                torch.load(
+                    os.path.join(
+                        self.parent.config.model_name_or_path, "scheduler.pt"
+                    )
+                )
+            )
+
+        if self.parent.config.fp16:
+            try:
+                from apex import amp
+            except ImportError:
+                raise ImportError(
+                    "Please install apex from https://www.github.com/nvidia/apex to use fp16 training."
+                )
+
+            self.model, optimizer = amp.initialize(
+                self.model,
+                optimizer,
+                opt_level=self.parent.config.fp16_opt_level,
+            )
+
+        # multi-gpu training
+        if self.parent.config.n_gpu > 1:
+            self.model = torch.nn.DataParallel(self.model)
+
+        # Distributed training
+        if self.parent.config.local_rank != -1:
+            self.model = torch.nn.parallel.DistributedDataParallel(
+                self.model,
+                device_ids=[self.parent.config.local_rank],
+                output_device=self.parent.config.local_rank,
+                find_unused_parameters=True,
+            )
+
+        # Train
+        logger.info("***** Running training *****")
+        logger.info("  Num examples = %d", len(train_dataset))
+        logger.info("  Num Epochs = %d", self.parent.config.num_train_epochs)
+        logger.info(
+            "  Instantaneous batch size per GPU = %d",
+            self.parent.config.per_gpu_train_batch_size,
+        )
+        logger.info(
+            "  Total train batch size (w. parallel, distributed & accumulation) = %d",
+            self.parent.config.train_batch_size
+            * self.parent.config.gradient_accumulation_steps
+            * (
+                torch.distributed.get_world_size()
+                if self.parent.config.local_rank != -1
+                else 1
+            ),
+        )
+        logger.info(
+            "  Gradient Accumulation steps = %d",
+            self.parent.config.gradient_accumulation_steps,
+        )
+        logger.info("  Total optimization steps = %d", t_total)
+
+        global_step = 1
+        epochs_trained = 0
+        steps_trained_in_current_epoch = 0
+        # Check if continuing training from a checkpoint
+        if os.path.exists(self.parent.config.model_name_or_path):
+            try:
+                # set global_step to gobal_step of last saved checkpoint from model path
+                checkpoint_suffix = self.parent.config.model_name_or_path.split(
+                    "-"
+                )[
+                    -1
+                ].split(
+                    "/"
```

Comment:
This looks like it assumes UNIX paths, can we use `pathlib.Path` here instead to make sure this will still work on Windows?